posthog/models/organization.py

```diff
@@ -191,18 +193,27 @@ def validate(self, *, user: Any = None, email: Optional[str] = None) -> None:
         _email = email or (hasattr(user, "email") and user.email)
 
         if _email and _email != self.target_email:
-            raise ValueError("This invite is intended for another email address.")
+            raise exceptions.ValidationError(
+                f"This invite is intended for another email address ({mask_email_address(self.target_email)}).",
+                code="invalid_recipient",
+            )
 
         if self.is_expired():
-            raise ValueError("This invite has expired. Please ask your admin for a new one.")
+            raise exceptions.ValidationError(
+                "This invite has expired. Please ask your admin for a new one.", code="expired",
+            )
 
         if OrganizationMembership.objects.filter(organization=self.organization, user=user).exists():
-            raise ValueError("User already is a member of the organization.")
+            raise exceptions.ValidationError(
+                "User already is a member of the organization.", code="user_already_member",
```

Comment:
If this is user-facing, may want to change to "You are already…". Though this is definitely an edge case, as normally no invite allowing this will exist.